cmake/try-sincospi.cc

```diff
@@ -0,0 +1,33 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file try-sincospi.cc
+//---------------------------------------------------------------------------//
+
+#include <cmath>
+
+#ifndef CELERITAS_SINCOSPI_PREFIX
+#    error "Expected CELERITAS_SINCOSPI_PREFIX to be defined"
+#endif
+
+#define CELER_CONCAT(PREFIX, FUNC) PREFIX##FUNC
```

Comment:
If it is important that those macro stays in sync with future changes in the original, we may want to note where they come from.